src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxDocumentConverterBase.cs

```diff
@@ -203,7 +204,17 @@ public IndexFieldsMapping GetKnownFields()
                 }
 
                 return;
-
+            
+            case ValueType.DateOnly:
+                var dateOnly = ((DateOnly)value).ToString(DefaultFormat.DateOnlyFormatToWrite, CultureInfo.InvariantCulture);
+                scope.Write(field.Id, dateOnly, ref entryWriter);
+                return;
+            
+            case ValueType.TimeOnly:
```

Comment:
Are there any other possible places where we could miss the handling of DateOnly / TimeOnly in Corax integration layer?

Reply:
No. In Lucene, we are checking if the term is representing time and when it is we look for field_Time in the index. Due to the lack of dynamic fields in Corax, we cannot achieve that at this point. So every operation with time is treated as a string comparison.

Reply:
Note that date compare is a *real* big concern for us in terms of query performance. We have a bunch of options on how to optimize that down the line. One of them is about not even checking the terms directly, only reading that from the entries.

Reply:
` One of them is about not even checking the terms directly, only reading that from the entries.`

Do you mean storing date binary in entry and checking it when doing query?

Reply:
Yes, correct. In the `AndWith`, we'll something like:

```
foreach(var itemId in items)
{
    var entry = GetEntry(itemId);
    if(entry.GetField("Date") > val) {
         yield itemId;
   }
}
```

Reply:
That is for later, however, once we are mostly done with the Corax work and can work on query optimizations

Reply:
I was talking with @redknightlois about some optimizations in UnaryMatch.
even we can do something like this:

```
foreach (var term in terms)
{
    if (term('Date') > val)
    {
        yield term.GetMatchingEntries();
    }
}

Reply:
Consider this query:

`from Posts where Tag = $tag and PostedAt < $now`

Assume we have millions of posts, most of them in the past, but the tag we are using is small.
So the left hand is ~100s of items, but on the right, we have to do union with millions.

Expensive to do that. Better to check the entry directly, instead.

Reply:
Ok, that makes sense.